### DISCUSSION TITLE
MEAN Stack 2.0

### DISCUSSION THREAD

**Discussion by mahenrique94** [frontendbr/forum#536] [Perguntas e Respostas]
Fala galera, tenho um projeto feito com MEAN Stack na primeira versão, com JS puro, estou querendo migra-lo para a versão nova com TS e Angular 2, alguém ja fez isso ? Não estou achando nenhum material na net.

Abraçooos

**Comment by ericdouglas** [comment]
@mahenrique94 o que você quer dizer com migrar?

Um detalhe é que o Angular está na [versão **4**](http://angularjs.blogspot.com.br/2017/03/angular-400-now-available.html) agora.

**Comment by fabioassuncao** [comment]
Cara, eu não cheguei a trabalhar em upgrade, mas suponho que você esteja se
referindo ao MEAN Stack com Angular 1.x; nesse caso, um refactor para 2.x
pode ser um pouco doloroso. Vale lembrar que a versão 2 não é apenas uma
atualização do código da 1.x, mas uma reescrita de toda a estrutura e
incluí uma série breaking changes. Eu ousaria dizer que ao invés de fazer
um refactor, talvez fosse interessante reescrever do zero — pelo menos o
que envolve o Angular — do que simplesmente refatorar o que já tem.

Alguns links que podem te ajudar:

https://www.ng-book.com/2/p/Converting-an-Angular-1-App-to-Angular-2/
<https://mailtrack.io/trace/link/e1b2acace37ed70ec2610c007108f85b58b10349?url=https%3A%2F%2Fwww.ng-book.com%2F2%2Fp%2FConverting-an-Angular-1-App-to-Angular-2%2F&userId=75874&signature=55cf95a2266e8761>

https://medium.com/@vascofernandes/the-road-to-angular-2-refactoring-angularjs-1-x-applications-for-smoother-upgrade-69c91364bc4c
<https://mailtrack.io/trace/link/b5c3c8dace3556ec1dba460a70a5a0ce69d55e22?url=https%3A%2F%2Fmedium.com%2F%40vascofernandes%2Fthe-road-to-angular-2-refactoring-angularjs-1-x-applications-for-smoother-upgrade-69c91364bc4c&userId=75874&signature=7a1d2c34b2f45c36>


  <https://mailtrack.io/> Enviado com Mailtrack
<https://mailtrack.io/install?source=signature&lang=pt&referral=fabio23gt@gmail.com&idSignature=24>
<#>

**Comment by mahenrique94** [comment]
Acabei explicando errado, quero refazer a mesma aplicação em MEAN 2.0, onde os códigos serão todos em TS(node, express, mongo, angular, etc...). Mas não acho nenhum tutorial na web sobre isso.

Teria que começar configurando o NodeJs com Typescript, depois configurar o express, realizar a conexão com o mongo e perfil o Angular 2 ou 4, tudo com TS e os mais novos recursos.

**Comment by felipefialho** [comment]
@suissa
